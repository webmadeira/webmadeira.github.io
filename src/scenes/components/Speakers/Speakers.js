import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import AppContainer from '../../../components/AppContainer/AppContainer'
import { H2 } from '../../../components/typography'
import Speaker from './components/Speaker/Speaker'

const Root = styled.section`
  padding-top: 80px;
  padding-bottom: 80px;
`

const Title = styled.section`
  text-align: left;
  color: ${props => props.theme.color('secondary', 'dark')};
  margin-bottom: 40px;
`

const List = styled.ul`
  display: grid;
  grid-gap: 20px;
  grid-template-columns: repeat(3, auto);
  
  ${({ theme }) => theme.mixins.mediaQuery.tablet`
    grid-gap: 10px;
  `};

  ${({ theme }) => theme.mixins.mediaQuery.phone`
    grid-template-columns: repeat(2, auto);
  `};
`

const ListItem = styled.li`
  list-style: none;
`

const renderSpeakers = speakers => speakers.map(speaker => (
  <ListItem key={speaker.id}>
    <Speaker speaker={speaker} />
  </ListItem>
))

const Speakers = ({ speakers }) => (
  <AppContainer>
    <Root>
      <Title>
        <H2>Speakers</H2>
      </Title>
      <List>
        {renderSpeakers(speakers)}
      </List>
    </Root>
  </AppContainer>
)

Speakers.propTypes = {
  speakers: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
}

export default Speakers
