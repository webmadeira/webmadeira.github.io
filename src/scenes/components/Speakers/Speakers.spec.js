import React from 'react'
import { shallow } from 'enzyme'
import Speakers from './Speakers'

describe('Speakers component', () => {
  it('must render properly', () => {
    const speakers = [
      {
        id: 0,
        name: 'André Gonçalves',
        company: 'Freelancer',
        jobTitle: 'Web Dev Consultant',
        personalWebsite: 'https://github.com/andregoncalves',
        bwPhoto: 'https://avatars3.githubusercontent.com/u/1074?s=460&v=4',
        photo: 'https://avatars3.githubusercontent.com/u/1074?s=460&v=4',
      },
      {
        id: 1,
        name: 'Aurélio Pita',
        company: 'Blip',
        jobTitle: 'Front End Developer',
        jobWebsite: 'https://blip.pt/',
        personalWebsite: 'https://github.com/jaapaurelio',
        bwPhoto: 'https://avatars3.githubusercontent.com/u/2837728?s=460&v=4',
        photo: 'https://avatars3.githubusercontent.com/u/2837728?s=460&v=4',
      },
    ]

    const component = shallow(<Speakers speakers={speakers} />)

    expect(component).toMatchSnapshot()
  })
})
