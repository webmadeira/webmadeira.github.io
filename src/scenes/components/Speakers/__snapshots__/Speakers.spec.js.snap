// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Speakers component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Speakers
    speakers={
      Array [
        Object {
          "bwPhoto": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
          "company": "Freelancer",
          "id": 0,
          "jobTitle": "Web Dev Consultant",
          "name": "André Gonçalves",
          "personalWebsite": "https://github.com/andregoncalves",
          "photo": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
        },
        Object {
          "bwPhoto": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
          "company": "Blip",
          "id": 1,
          "jobTitle": "Front End Developer",
          "jobWebsite": "https://blip.pt/",
          "name": "Aurélio Pita",
          "personalWebsite": "https://github.com/jaapaurelio",
          "photo": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.section>
        <styled.section>
          <styled.h2>
            Speakers
          </styled.h2>
        </styled.section>
        <styled.ul>
          <styled.li>
            <Speaker
              speaker={
                Object {
                  "bwPhoto": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                  "company": "Freelancer",
                  "id": 0,
                  "jobTitle": "Web Dev Consultant",
                  "name": "André Gonçalves",
                  "personalWebsite": "https://github.com/andregoncalves",
                  "photo": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                }
              }
            />
          </styled.li>
          <styled.li>
            <Speaker
              speaker={
                Object {
                  "bwPhoto": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                  "company": "Blip",
                  "id": 1,
                  "jobTitle": "Front End Developer",
                  "jobWebsite": "https://blip.pt/",
                  "name": "Aurélio Pita",
                  "personalWebsite": "https://github.com/jaapaurelio",
                  "photo": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                }
              }
            />
          </styled.li>
        </styled.ul>
      </styled.section>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.section>
            <styled.h2>
              Speakers
            </styled.h2>
          </styled.section>,
          <styled.ul>
            <styled.li>
              <Speaker
                speaker={
                  Object {
                    "bwPhoto": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                    "company": "Freelancer",
                    "id": 0,
                    "jobTitle": "Web Dev Consultant",
                    "name": "André Gonçalves",
                    "personalWebsite": "https://github.com/andregoncalves",
                    "photo": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                  }
                }
              />
            </styled.li>
            <styled.li>
              <Speaker
                speaker={
                  Object {
                    "bwPhoto": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                    "company": "Blip",
                    "id": 1,
                    "jobTitle": "Front End Developer",
                    "jobWebsite": "https://blip.pt/",
                    "name": "Aurélio Pita",
                    "personalWebsite": "https://github.com/jaapaurelio",
                    "photo": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                  }
                }
              />
            </styled.li>
          </styled.ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.h2>
              Speakers
            </styled.h2>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Speakers",
            },
            "ref": null,
            "rendered": "Speakers",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.li>
                <Speaker
                  speaker={
                    Object {
                      "bwPhoto": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                      "company": "Freelancer",
                      "id": 0,
                      "jobTitle": "Web Dev Consultant",
                      "name": "André Gonçalves",
                      "personalWebsite": "https://github.com/andregoncalves",
                      "photo": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                    }
                  }
                />
              </styled.li>,
              <styled.li>
                <Speaker
                  speaker={
                    Object {
                      "bwPhoto": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                      "company": "Blip",
                      "id": 1,
                      "jobTitle": "Front End Developer",
                      "jobWebsite": "https://blip.pt/",
                      "name": "Aurélio Pita",
                      "personalWebsite": "https://github.com/jaapaurelio",
                      "photo": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                    }
                  }
                />
              </styled.li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "children": <Speaker
                  speaker={
                    Object {
                      "bwPhoto": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                      "company": "Freelancer",
                      "id": 0,
                      "jobTitle": "Web Dev Consultant",
                      "name": "André Gonçalves",
                      "personalWebsite": "https://github.com/andregoncalves",
                      "photo": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                    }
                  }
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "speaker": Object {
                    "bwPhoto": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                    "company": "Freelancer",
                    "id": 0,
                    "jobTitle": "Web Dev Consultant",
                    "name": "André Gonçalves",
                    "personalWebsite": "https://github.com/andregoncalves",
                    "photo": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "children": <Speaker
                  speaker={
                    Object {
                      "bwPhoto": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                      "company": "Blip",
                      "id": 1,
                      "jobTitle": "Front End Developer",
                      "jobWebsite": "https://blip.pt/",
                      "name": "Aurélio Pita",
                      "personalWebsite": "https://github.com/jaapaurelio",
                      "photo": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                    }
                  }
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "speaker": Object {
                    "bwPhoto": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                    "company": "Blip",
                    "id": 1,
                    "jobTitle": "Front End Developer",
                    "jobWebsite": "https://blip.pt/",
                    "name": "Aurélio Pita",
                    "personalWebsite": "https://github.com/jaapaurelio",
                    "photo": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.section>
          <styled.section>
            <styled.h2>
              Speakers
            </styled.h2>
          </styled.section>
          <styled.ul>
            <styled.li>
              <Speaker
                speaker={
                  Object {
                    "bwPhoto": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                    "company": "Freelancer",
                    "id": 0,
                    "jobTitle": "Web Dev Consultant",
                    "name": "André Gonçalves",
                    "personalWebsite": "https://github.com/andregoncalves",
                    "photo": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                  }
                }
              />
            </styled.li>
            <styled.li>
              <Speaker
                speaker={
                  Object {
                    "bwPhoto": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                    "company": "Blip",
                    "id": 1,
                    "jobTitle": "Front End Developer",
                    "jobWebsite": "https://blip.pt/",
                    "name": "Aurélio Pita",
                    "personalWebsite": "https://github.com/jaapaurelio",
                    "photo": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                  }
                }
              />
            </styled.li>
          </styled.ul>
        </styled.section>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.section>
              <styled.h2>
                Speakers
              </styled.h2>
            </styled.section>,
            <styled.ul>
              <styled.li>
                <Speaker
                  speaker={
                    Object {
                      "bwPhoto": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                      "company": "Freelancer",
                      "id": 0,
                      "jobTitle": "Web Dev Consultant",
                      "name": "André Gonçalves",
                      "personalWebsite": "https://github.com/andregoncalves",
                      "photo": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                    }
                  }
                />
              </styled.li>
              <styled.li>
                <Speaker
                  speaker={
                    Object {
                      "bwPhoto": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                      "company": "Blip",
                      "id": 1,
                      "jobTitle": "Front End Developer",
                      "jobWebsite": "https://blip.pt/",
                      "name": "Aurélio Pita",
                      "personalWebsite": "https://github.com/jaapaurelio",
                      "photo": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                    }
                  }
                />
              </styled.li>
            </styled.ul>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <styled.h2>
                Speakers
              </styled.h2>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Speakers",
              },
              "ref": null,
              "rendered": "Speakers",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.li>
                  <Speaker
                    speaker={
                      Object {
                        "bwPhoto": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                        "company": "Freelancer",
                        "id": 0,
                        "jobTitle": "Web Dev Consultant",
                        "name": "André Gonçalves",
                        "personalWebsite": "https://github.com/andregoncalves",
                        "photo": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                      }
                    }
                  />
                </styled.li>,
                <styled.li>
                  <Speaker
                    speaker={
                      Object {
                        "bwPhoto": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                        "company": "Blip",
                        "id": 1,
                        "jobTitle": "Front End Developer",
                        "jobWebsite": "https://blip.pt/",
                        "name": "Aurélio Pita",
                        "personalWebsite": "https://github.com/jaapaurelio",
                        "photo": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                      }
                    }
                  />
                </styled.li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "class",
                "props": Object {
                  "children": <Speaker
                    speaker={
                      Object {
                        "bwPhoto": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                        "company": "Freelancer",
                        "id": 0,
                        "jobTitle": "Web Dev Consultant",
                        "name": "André Gonçalves",
                        "personalWebsite": "https://github.com/andregoncalves",
                        "photo": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                      }
                    }
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "speaker": Object {
                      "bwPhoto": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                      "company": "Freelancer",
                      "id": 0,
                      "jobTitle": "Web Dev Consultant",
                      "name": "André Gonçalves",
                      "personalWebsite": "https://github.com/andregoncalves",
                      "photo": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "class",
                "props": Object {
                  "children": <Speaker
                    speaker={
                      Object {
                        "bwPhoto": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                        "company": "Blip",
                        "id": 1,
                        "jobTitle": "Front End Developer",
                        "jobWebsite": "https://blip.pt/",
                        "name": "Aurélio Pita",
                        "personalWebsite": "https://github.com/jaapaurelio",
                        "photo": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                      }
                    }
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "speaker": Object {
                      "bwPhoto": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                      "company": "Blip",
                      "id": 1,
                      "jobTitle": "Front End Developer",
                      "jobWebsite": "https://blip.pt/",
                      "name": "Aurélio Pita",
                      "personalWebsite": "https://github.com/jaapaurelio",
                      "photo": "https://avatars3.githubusercontent.com/u/2837728?s=460&v=4",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
